src/strval.rs

```diff
@@ -61,12 +63,24 @@ pub fn from_string(s: &str, t: &ScSpecTypeDef) -> Result<ScVal, StrValError> {
             match serde_json::from_str(s) {
                 // First, see if it is a json array
                 Ok(Value::Array(raw)) => from_json(&Value::Array(raw), t)?,
-                // Not a json array, just grab the bytes.
-                _ => ScVal::Object(Some(ScObject::Binary(
-                    s.as_bytes()
-                        .try_into()
-                        .map_err(|_| StrValError::InvalidValue)?,
-                ))),
+                _ =>
+                // it could be a G- strkey
+                {
+                    match StrkeyPublicKeyEd25519::from_string(s) {
+                        Ok(key) => {
+                            let b = &key.0;
+                            ScVal::Object(Some(ScObject::Binary(
```

Comment:
Shouldn't this be an ScPublicKey? Or maybe that type is missing from the ScSpecTypeDef?

Reply:
I am using `stellar_strkey::StrkeyPublicKeyEd25519` I can't find `ScPublicKey`.

Reply:
ScPublicKey and ScHash aren't being used anywhere at the moment, even within the host. All the examples that do have keys use `Binary`.